### PR TITLE
Add PollingTime & MinimumIdleTime to BackgroundServiceOptions

### LINKED ISSUE #17031: Make the polling time and min idle time of `ModularBackgroundService` configurable
### Is your feature request related to a problem?

`ModularBackgroundService` executes background tasks in one-minute batches:

https://github.com/OrchardCMS/OrchardCore/blob/0917712993664113585516bca6b06149c3af7031/src/OrchardCore/OrchardCore/Modules/ModularBackgroundService.cs#L19

Additionally, there's a 10-second idle time between batches:

https://github.com/OrchardCMS/OrchardCore/blob/0917712993664113585516bca6b06149c3af7031/src/OrchardCore/OrchardCore/Modules/ModularBackgroundService.cs#L20

Neither of these are configurable, which prevents customizing background execution, forcing you to wait at least 1m 10s for a background task. Our particular use case at hand is for UI testing, where we have to wait for background tasks instead of being able to tell to execute them faster.

### Describe the solution you'd like

Have `IOptions` configuration for both values.

### Describe alternatives you've considered

If somehow these delays would use `IClock` internally then that would allow the UI testing use case too (since we can have a test-specific implementation that forwards time, [what we actually have](https://github.com/Lombiq/UI-Testing-Toolbox/blob/9f7ff0317606d4a5fa689e84f7d46d25fd25a520/Lombiq.Tests.UI.Samples/Tests/ShiftTimeTests.cs)). It wouldn't help in other cases, though.

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #17031

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Abstractions/Modules/BackgroundServiceOptions.cs [hunk 1]
```diff
@@ -3,4 +3,14 @@ namespace OrchardCore.Modules;
 public class BackgroundServiceOptions
 {
     public bool ShellWarmup { get; set; }
+
+    /// <summary>
+    /// Gets or sets the time in seconds between each background task execution.
+    /// </summary>
+    public int PollingTime { get; set; } = 60;
```
[the remark this answers is not in the document]
Comment (hishamco) [reply src/OrchardCore/OrchardCore.Abstractions/Modules/BackgroundServiceOptions.cs:10]: Ok, for that we need to make things consistent across the code base

File: src/OrchardCore.Cms.Web/appsettings.json [hunk 2]
```diff
@@ -135,7 +135,9 @@
     //},
     // By default background tasks are waiting for their shell to be lazily built on a first matching request.
     //"OrchardCore_BackgroundService": {
-    //  "ShellWarmup": true // Allows to eagerly build shell containers just before executing their first background task.
+    //  "ShellWarmup": true,
+    //  "PollingTime": "00:01:00",
```
Comment (Piedone) [inline src/OrchardCore.Cms.Web/appsettings.json:139]:
```suggestion
    // The polling time for the background tasks, i.e. how much to wait between executing all the background tasks of a given tenant.
    //  "PollingTime": "00:01:00",
```

File: src/OrchardCore/OrchardCore.Abstractions/Modules/BackgroundServiceOptions.cs [hunk 3]
```diff
@@ -2,5 +2,18 @@ namespace OrchardCore.Modules;
 
 public class BackgroundServiceOptions
 {
+    /// <summary>
+    /// Gets or sets a value indicating whether the shell should be warmed up.
+    /// </summary>
     public bool ShellWarmup { get; set; }
+
+    /// <summary>
+    /// Gets or sets the polling time for the background task.
```
Comment (Piedone) [inline src/OrchardCore/OrchardCore.Abstractions/Modules/BackgroundServiceOptions.cs:11]:
```suggestion
    /// Gets or sets the polling time for the background tasks, i.e. how much to wait between executing all the background tasks of a given tenant.
```

File: src/OrchardCore/OrchardCore.Abstractions/Modules/BackgroundServiceOptions.cs [hunk 4]
```diff
@@ -2,5 +2,18 @@ namespace OrchardCore.Modules;
...
+    /// </summary>
+    public TimeSpan PollingTime { get; set; } = TimeSpan.FromMinutes(1);
+
+    /// <summary>
+    /// Gets or set the minimum idle time before the background task is triggered.
```
Comment (Piedone) [inline src/OrchardCore/OrchardCore.Abstractions/Modules/BackgroundServiceOptions.cs:16]:
```suggestion
    /// Gets or sets the minimum idle time before the background tasks of a tenant are triggered, similar to <see cref="PollingTime" />.
```

File: src/OrchardCore.Cms.Web/appsettings.json [hunk 5]
```diff
@@ -135,7 +135,9 @@
...
+    //  "MinimumIdelTime": "00:00:10"
```
Comment (Piedone) [inline src/OrchardCore.Cms.Web/appsettings.json:140]:
```suggestion
    // The minimum idle time before the background tasks of a tenant are triggered, similar to PollingTime.
    //  "MinimumIdelTime": "00:00:10"
```

File: src/OrchardCore/OrchardCore.Abstractions/Modules/BackgroundServiceOptions.cs [hunk 6]
```diff
@@ -2,5 +2,18 @@ namespace OrchardCore.Modules;
...
+    /// </summary>
+    public TimeSpan MinimumIdelTime { get; set; } = TimeSpan.FromSeconds(10);
```
Comment (Piedone) [inline src/OrchardCore/OrchardCore.Abstractions/Modules/BackgroundServiceOptions.cs:18]:
```suggestion
    public TimeSpan MinimumIdleTime { get; set; } = TimeSpan.FromSeconds(10);
```

**Review by Piedone — CHANGES_REQUESTED** [review]
Please add a note about this to the release notes.

File: src/OrchardCore/OrchardCore/Modules/ModularBackgroundService.cs [hunk 7]
```diff
@@ -66,7 +63,7 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
         {
             try
             {
-                await Task.Delay(_minIdleTime, stoppingToken);
+                await Task.Delay(_options.MinimumIdelTime, stoppingToken);
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore/Modules/ModularBackgroundService.cs:66]:
```suggestion
                await Task.Delay(_options.MinimumIdleTime, stoppingToken);
```

File: src/OrchardCore/OrchardCore/Modules/ModularBackgroundService.cs [hunk 8]
```diff
@@ -336,11 +333,11 @@ await shellScope.UsingAsync(async scope =>
         });
     }
 
-    private static async Task WaitAsync(Task pollingDelay, CancellationToken stoppingToken)
+    private async Task WaitAsync(Task pollingDelay, CancellationToken stoppingToken)
     {
         try
         {
-            await Task.Delay(_minIdleTime, stoppingToken);
+            await Task.Delay(_options.MinimumIdelTime, stoppingToken);
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore/Modules/ModularBackgroundService.cs:340]:
```suggestion
            await Task.Delay(_options.MinimumIdleTime, stoppingToken);
```

File: src/OrchardCore/OrchardCore.Abstractions/Modules/BackgroundServiceOptions.cs [hunk 9]
```diff
@@ -8,12 +8,12 @@ public class BackgroundServiceOptions
...
 
     /// <summary>
-    /// Gets or sets the polling time for the background task.
+    /// Gets or sets the polling time for the background tasks, i.e. how much to wait between executing all the background tasks of a given tenant.
     /// </summary>
     public TimeSpan PollingTime { get; set; } = TimeSpan.FromMinutes(1);
 
     /// <summary>
-    /// Gets or set the minimum idle time before the background task is triggered.
+    /// Gets or sets the minimum idle time before the background tasks of a tenant are triggered, similar to <see cref="PollingTime" />.
```
Comment (Piedone) [inline src/OrchardCore/OrchardCore.Abstractions/Modules/BackgroundServiceOptions.cs:16]:
```suggestion
    /// Gets or sets the minimum idle time before the background tasks of a tenant are triggered, as well as between tasks similar to <see cref="PollingTime" />.
```

File: src/OrchardCore.Cms.Web/appsettings.json [hunk 10]
```diff
@@ -136,7 +136,9 @@
     // By default background tasks are waiting for their shell to be lazily built on a first matching request.
     //"OrchardCore_BackgroundService": {
     //  "ShellWarmup": true,
+    // The polling time for the background tasks, i.e. how much to wait between executing all the background tasks of a given tenant.
     //  "PollingTime": "00:01:00",
+    // The minimum idle time before the background tasks of a tenant are triggered, similar to PollingTime.
```
Comment (Piedone) [inline src/OrchardCore.Cms.Web/appsettings.json:141]:
```suggestion
    // The minimum idle time before the background tasks of a tenant are triggered, as well as between tasks similar to PollingTime.
```

**Review by Piedone — CHANGES_REQUESTED** [review]
Note I updated the previous code suggestions a bit ago.

File: src/OrchardCore.Cms.Web/appsettings.json [hunk 11]
```diff
@@ -136,7 +136,9 @@
...
     //  "MinimumIdelTime": "00:00:10"
```
Comment (Piedone) [inline src/OrchardCore.Cms.Web/appsettings.json:142]:
```suggestion
    //  "MinimumIdleTime": "00:00:10"
```

**Comment by hishamco** [comment]
Working from mobile is a hard, I will check it tomorrow from with my laptop

**Review by Piedone — APPROVED** [review]

**Pull request merged**